sklearn/preprocessing/_label.py

```diff
@@ -524,16 +528,18 @@ def label_binarize(y, *, classes, neg_label=0, pos_label=1, sparse_output=False)
     if y_type == "unknown":
         raise ValueError("The type of target data is not known")
 
-    n_samples = y.shape[0] if sp.issparse(y) else len(y)
-    n_classes = len(classes)
+    n_samples = y.shape[0] if hasattr(y, "shape") else len(y)
+    n_classes = classes.shape[0] if hasattr(classes, "shape") else len(classes)
     classes = np.asarray(classes)
```

Comment:
This will not work if `classes` a GPU hosted torch tensor. It will fail similarly for CuPy arrays (I think). We need to use `_convert_numpy` for now if we want to move that to numpy/CPU.